File: Array/26_remove_duplicates_from_sorted_array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i, j = 0, 1
        n = len(nums)

        for j in range(1, n):
            if nums[i] == nums[j]:
                continue
            nums[i + 1] = nums[j]
            i += 1

        return i + 1

File: Array/26_remove_duplicates_from_sorted_array.cpp
class Solution {
    public:
        int removeDuplicates(vector<int>& nums) {
            int left = 1;
    
            for(int right = 1; right < nums.size(); right += 1) {
                if(nums[right] != nums[right - 1]) {
                    nums[left++] = nums[right];
                }
            }
    
            return left;
        }
    };

// revised - 02/12/2025